File: trunk/Yet another programm-introduction to the new semester/Yaskov/lexer/Token.java
/*
 *
 * "Простейший транслятор"
 *
 * (c) Яськов Сергей, 261, 2010;
 *
 * представитель списка токенов, создаваемого интерпретатором. у каждого токена есть тип, у некоторых из них -
 * атрибут;
 *
 */

package lexer;

public class Token {
    private TokenType tokenType;
    private int attribute;

    Token(TokenType token, int attribute) {
        this.tokenType = token;
        this.attribute = attribute;
    }

    public TokenType getType() {
        return tokenType;
    }

    public int getAttribute() {
        return attribute;
    }
}

File: Interpreter (with let, if, GUI etc)/Bojarovski/Token.cs
/*
 * (c) Stefan Bojarovski 2010
 * */
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ASTBuilder
{
    class Token
    {
        private String Value;
        private TokenType Type;

        public Token(String someString, TokenType someType)
        {
            Value = someString;
            Type = someType;
        }

        public String getValue()
        {
            return Value;
        }
        public void setValue(String someString)
        {
            Value = someString;
        }
        public TokenType getType()
        {
            return Type;
        }
        public void print(StreamWriter some_str)
        {
            if (Type != TokenType.WHITESPACE)
            {
                some_str.WriteLine("Token value: " + Value + " Token type: " + Type.ToString());
            }
        }
    }
}
